eng/common/TestResources/New-TestResources.ps1

```diff
@@ -580,14 +580,18 @@ try {
     $PSBoundParameters['TestApplicationOid'] = $TestApplicationOid
     $PSBoundParameters['TestApplicationSecret'] = $TestApplicationSecret
 
-    # Grant the test service principal ownership over the resource group. This may fail if the provisioner is a
-    # service principal without permissions to grant RBAC roles to other service principals. That should not be
-    # considered a critical failure, as the test application may have subscription-level permissions and not require
-    # the explicit grant.
-    #
-    # Ignore this check if $AzureTestPrincipal is specified as role assignment will already have been attempted on a
-    # previous run, and these error messages can be misleading for local runs.
-    if (!$resourceGroupRoleAssigned -and !$AzureTestPrincipal) {
+    # If the role hasn't been explicitly assigned to the resource group and a cached service principal is in use,
+    # query to see if the grant is needed.
+    if (!$resourceGroupRoleAssigned -and $AzureTestPrincipal) {
```

Comment:
Why do we need the condition? Can't we always run the New-AzRoleAssignment ?

Reply:
We had it that way at one point and received feedback that the messaging from the cmdlet output looked like an error when granting a role that was already assigned; people were confused and some of the reviewers felt strongly about cleaning that up - so we conditionalized it.

Reply:
Will this Get-AzRoleAssignment call pick up assignments from a subscription level? It might help us avoid an error message for the `the test application may have subscription-level permissions and not require the explicit grant.`

Reply:
It does pick up inherited permissions, so far as I observed in local testing.